src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -181,18 +192,22 @@ impl TimeTrigger {
             let now: std::time::Duration = SystemTime::now()
                 .duration_since(UNIX_EPOCH)
                 .expect("system time before Unix epoch");
-            NaiveDateTime::from_timestamp_opt(now.as_secs() as i64, now.subsec_nanos())
+            DateTime::from_timestamp(now.as_secs() as i64, now.subsec_nanos())
                 .unwrap()
+                .naive_local()
                 .and_local_timezone(Local)
                 .unwrap()
         };
 
         #[cfg(not(test))]
         let current = Local::now();
-        let next_time = TimeTrigger::get_next_time(current, config.interval, config.modulate);
+        let next_time =
+            TimeTrigger::get_next_time(current, config.interval, config.modulate).unwrap();
         let next_roll_time = if config.max_random_delay > 0 {
             let random_delay = rand::thread_rng().gen_range(0..config.max_random_delay);
-            next_time + Duration::seconds(random_delay as i64)
+            next_time
+                + Duration::try_seconds(random_delay as i64)
+                    .unwrap_or(Duration::try_milliseconds(i64::MAX).unwrap())
```

Comment:
need to remove this unwrap and handle the error

Reply:
I'm assuming you're referring to the unwrap on `Duration::try_milliseconds(i64::MAX).unwrap()` . That's not a Result but an Option with a known valid value being passed in. I'll add a comment saying as much.

Reply:
Dont we want to alert the user if there was an error here?

Reply:
I use the `Duration::try_milliseconds(i64::MAX).unwrap()` without alerting users because:
1. The users unlikely set `max_random_delay` exceeding the upper limit. 
2. When users indeed set a random time exceeding the upper limit, it's not against the users' wishes to reduce it to `i64::MAX` milliseconds.
3.  It's hard to handle error here. As we use a random time with `max_random_delay`, it will randomly return errors too, which is more likely to confuse users.

If we handle to process this error, including the error in `get_next_time`, maybe we can add a new `check_config` method to troubleshoot incorrect configurations before `new()` called, IMO.

Reply:
Added a comment to the effect of bullet 2.